ledger/ledger.go

```diff
@@ -137,14 +137,22 @@ func OpenLedger(
 }
 
 func (l *Ledger) reloadLedger() error {
-	// close first.
-	l.trackers.close()
+	// similar to the Close function, we want to start by closing the blockQ first. The
+	// blockQ is having a sync goroutine which indirectly calls other trackers. We want to eliminate that go-routine first,
+	// and follow up by taking the trackers lock.
 	if l.blockQ != nil {
 		l.blockQ.close()
 		l.blockQ = nil
 	}
 
-	// reload.
+	// take the trackers lock. This would ensure that no other goroutine is using the trackers.
+	l.trackerMu.Lock()
```

Comment:
I was thinking about it - the gains in doing this are pretty minor since it's only 5-6 lines of code. unifying the code path here is mostly cosmetic, and I think that proper in-code documentation would be just as beneficial.